Add zipaction integration test cases.

lzbj opened the pull request:
Address issue #245.

lzbj commented:
@houshengbo , please help review, it addressed issue 245 as you opened, please notice maybe later we'll change the behavior, because now the user needs to install the deps manually, we plan to support the automatic install deps. As issue https://github.com/openwhisk/openwhisk-wskdeploy/issues/250 mentioned.

houshengbo commented:
@lzbj Do we actually need the files under node_modules/left-pad? 
You submitted a lot of added packages, which makes me confused. Are they really necessary? This PR is supposed to fix the zip action, but where is the zip file?
I personally do not have a positive feeling about this left-pad package. There is even the "F" word inside some of the files.

houshengbo commented on tests/src/integration/zipaction/actions/padding/node_modules/left-pad/.npmignore:
```diff
@@ -0,0 +1,2 @@
+node_modules
+npm-debug.log
```
Do we need this file?

houshengbo commented on tests/src/integration/zipaction/actions/padding/node_modules/left-pad/.travis.yml:
```diff
@@ -0,0 +1,6 @@
+ - "6"
+ - "5"
+ - "4"
+ - "0.12"
```
Do we need this file?

houshengbo commented on tests/src/integration/zipaction/actions/padding/node_modules/left-pad/COPYING:
```diff
@@ -0,0 +1,14 @@
+
+            DO WHAT THE FUCK YOU WANT TO PUBLIC LICENSE
+   TERMS AND CONDITIONS FOR COPYING, DISTRIBUTION AND MODIFICATION
+
```
Do we need this file?

houshengbo commented on tests/src/integration/zipaction/actions/padding/node_modules/left-pad/README.md:
```diff
@@ -0,0 +1,36 @@
+## left-pad
```
Do we need this file?

houshengbo commented on tests/src/integration/zipaction/actions/padding/node_modules/left-pad/index.js:
```diff
@@ -0,0 +1,52 @@
+'use strict';
+module.exports = leftPad;
+
+var cache = [
```
Do we need this file?

houshengbo commented on tests/src/integration/zipaction/actions/padding/node_modules/left-pad/package.json:
```diff
@@ -0,0 +1,103 @@
+  "_args": [
+    [
+      {
+        "raw": "left-pad@1.1.3",
```
Do we need this file?

houshengbo commented on tests/src/integration/zipaction/actions/padding/node_modules/left-pad/perf/O(n).js:
```diff
@@ -0,0 +1,17 @@
+module.exports = function (str, len, ch) {
+  str = str + '';
+
+  len = len - str.length;
```
Do we need this file?

houshengbo commented on tests/src/integration/zipaction/actions/padding/node_modules/left-pad/perf/es6Repeat.js:
```diff
@@ -0,0 +1,13 @@
+'use strict';
+
+module.exports = function (str, len, ch) {
+  str = str + '';
```
Do we need this file?

houshengbo commented on tests/src/integration/zipaction/actions/padding/node_modules/left-pad/perf/perf.js:
```diff
@@ -0,0 +1,40 @@
+'use strict';
+var oN = require('./O(n)');
+var es6Repeat = require('./es6Repeat');
+var current = require('../');
```
Do we need this file?

houshengbo commented on tests/src/integration/zipaction/actions/padding/node_modules/left-pad/test.js:
```diff
@@ -0,0 +1,52 @@
+/* This program is free software. It comes without any warranty, to
+     * the extent permitted by applicable law. You can redistribute it
+     * and/or modify it under the terms of the Do What The Fuck You Want
```
Do we need this file?

houshengbo commented on tests/src/integration/zipaction/actions/padding/package.json:
```diff
@@ -0,0 +1,7 @@
+{
+  "name": "my-action",
+  "main": "index.js",
```
Do we need this file?

lzbj replied:
@houshengbo , I reply all your comments here: this is a sample use case from the openwhisk zip action doc, for zipped actions, yes, that's necessary, you can try by running npm install under the action.

lzbj commented:
@houshengbo @mrutkows , could you please have a review, I updated and removed some unnecessary files, except ones that sketch the node module hierarchy, sorry I didn't noticed there are some bad words, I deleted the file.

houshengbo commented:
@lzbj The package you are using is not appropriate, since the license of this package is weird and has inappropriate language in it. I suggest you switch to use another package.

lzbj commented:
@houshengbo , thanks for review, yes, I'll find another package with minimum footprint.

lzbj commented:
@houshengbo , I updated the referenced package and removed the unnecessary files, please help review.

lzbj commented:
@mrutkows @daisy-ycguo , could you please help review this pr, it seems Vincent is busy on other projects.

mrutkows commented on tests/src/integration/zipaction/deployment.yml:
```diff
@@ -0,0 +1,15 @@
+application:
+  name: wskdeploy-samples
+  namespace: guest
+  credential: 23bc46b1-71f6-4ed5-8c54-816aa4f8c502:123zO3xZCLrMN6v2BKK1dXYFpXlPkccOFqm12CdAsMgRU4VrNZ9lyGVCGuMDGIwP
```
@houshengbo is including a key here correct?

mrutkows commented on tests/src/integration/zipaction/zipaction_test.go:
```diff
@@ -22,6 +23,6 @@ func TestZipAction(t *testing.T) {
-	manifestPath   = os.Getenv("GOPATH") + "/src/github.com/apache/incubator-openwhisk-wskdeploy/tests/src/integration/zipaction/manifest.yaml"
-	deploymentPath = "" 
+	manifestPath   = os.Getenv("GOPATH") + "/src/github.com/openwhisk/openwhisk-wskdeploy/tests/src/integration/zipaction/manifest.yml"
+	deploymentPath = os.Getenv("GOPATH") + "/src/github.com/openwhisk/openwhisk-wskdeploy/tests/src/integration/zipaction/deployment.yml"
```
The path should reflect the new "apache" organization.

lzbj replied:
Yes, correct.

mrutkows commented on utils/misc.go:
```diff
@@ -192,6 +192,10 @@ func CreateFolderZip(src, des string) error {
 		writer, err := zipWritter.CreateHeader(header)
```
Please correct object name to "ZipWriter" (from "ZipWritter")

lzbj replied:
Ok, sure.

mrutkows commented on tests/src/integration/zipaction/manifest.yml:
```diff
@@ -0,0 +1,15 @@
+  actions:
+    cat:
+      version: 1.0
+      location: actions/cat
```
This is a good simple (in-line), single action test (better than previous one).  We should perhaps create another test that uses multiple (sub)directories.

lzbj replied:
Ok, I'll open another issue to track it.

mrutkows commented on utils/misc.go:
```diff
@@ -156,96 +156,59 @@ func GetJSONType(j interface{}) string {
-		Check(err)
-		return err
-	})
+func NewZipWritter(src, des string) *ZipWritter {
```
Should be "ZipWriter"

mrutkows commented on utils/misc.go:
```diff
@@ -156,96 +156,59 @@ func GetJSONType(j interface{}) string {
+}
 
-	return err
+type ZipWritter struct {
```
name should be "ZipWriter"

mrutkows commented on tests/src/integration/zipaction/actions/cat/index.js:
```diff
@@ -0,0 +1,5 @@
+function myAction(args) {
+    var cat = require('cat');
+    cat('https://baidu.com', console.log); 
```
Change this to a neutral website (like openwhisk.org) <or even better> do not require HTTPS and "cat" some local file?

mrutkows commented:
@houshengbo the new testcase looks good (simple

pritidesai commented:
Thanks a bunch David for this fix, I downloaded source from this pull request, built wskdeploy binary and tested it. I was able to create `cat-get-action` action as stated in #269 and run it successfully:

```
wsk action invoke cat-get/cat-get-action --blocking --param id 2
"response": {
        "result": {
            "color": "red",
            "id": 2,
            "name": "minnie"
        },
        "status": "success",
        "success": true
    },
```
